### ISSUE TITLE
Codekit wont compile with XY Grid

### ISSUE THREAD

**Issue by bgarrant** [zurb/foundation-sites#10300]
For some reason this will not compile when I enable the XY Grid imports?  Do I need to disable some others?

Error:

Libsass: Error: Properties are only allowed within rules, directives, mixin includes, or other properties.
        on line 1 of Applications/MAMP/htdocs/srmt-nsn.dev/scss/master.scss
>> // CUSTOM COLORS
   ^

```
// CUSTOM COLORS

$my-purple: #332D7B;
$my-dark-purple: #2B2669;
$my-red: #EF3C35;
$my-white: #FFFFFF;
$my-background-gray: #F6F6F6;
$my-border-gray: #CACACA;

// FOUNDATION FOR SITES
// ---------------------------------------------------------------------------
//@import '../node_modules/motion-ui/motion-ui';
@import 'settings';
@import '../foundation-sites/scss/foundation';

@include foundation-global-styles;
@include xy-grid;
@include foundation-xy-grid-classes;
// @include foundation-grid;
@include foundation-flex-grid;
@include foundation-flex-classes;
@include foundation-typography;
@include foundation-forms;
@include foundation-button;
@include foundation-accordion;
@include foundation-accordion-menu;
@include foundation-badge;
@include foundation-breadcrumbs;
@include foundation-button-group;
@include foundation-callout;
@include foundation-card;
@include foundation-close-button;
@include foundation-menu;
@include foundation-menu-icon;
@include foundation-drilldown-menu;
@include foundation-dropdown;
@include foundation-dropdown-menu;
@include foundation-responsive-embed;
@include foundation-label;
@include foundation-media-object;
@include foundation-off-canvas;
@include foundation-orbit;
@include foundation-pagination;
@include foundation-progress-bar;
@include foundation-slider;
@include foundation-sticky;
@include foundation-reveal;
@include foundation-switch;
@include foundation-table;
@include foundation-tabs;
@include foundation-thumbnail;
@include foundation-title-bar;
@include foundation-tooltip;
@include foundation-top-bar;
@include foundation-visibility-classes;
// @include foundation-float-classes;

// CUSTOM STYLESHEET
// ---------------------------------------------------------------------------
@import "_custom";
```

**Comment by bgarrant** [comment]
do I need to disable these or something?

```
@include foundation-flex-grid;
@include foundation-flex-classes;
```

**Comment by kball** [comment]
You shouldn't need to disable those... hmm... do you have more of a backtrace you can share?

It may be related to https://github.com/zurb/foundation-sites/issues/10279

We're going to turn around a patch release tomorrow that has fixes to the above.

**Comment by bgarrant** [comment]
The issue is definitely in `@include xy-grid;`.  If I comment that one out it works

**Comment by kball** [comment]
Hmm... as a way to narrow in on if it's likely the same issue, can you try doing

`@include foundation-xy-grid-classes($vertical-grid: false)`

and see if it still errors?

**Comment by bgarrant** [comment]
Yes that works fine that way.  Compiles as it should

**Comment by kball** [comment]
Ok, I'm almost 100% sure this is due to the same underlying issue as #10279 and is fixed with https://github.com/zurb/foundation-sites/pull/10286

We'll have a patch update out tomorrow with that fix in. I'll leave this open until we're able to confirm

**Comment by bgarrant** [comment]
Great.  Thank you.

**Comment by bgarrant** [comment]
I tried that fix and it did not seem to help.  We will try it again tomorrow after you  put out the patch.

**Comment by kudlatyamroth** [comment]
I think its by @include xy-grid, as this mixins is:
`@mixin xy-grid(
  $direction: horizontal,
  $wrap: true
) {
  $direction: if($direction == 'horizontal', row, column);
  $wrap: if($wrap, wrap, nowrap);

  display: flex;
  flex-flow: $direction $wrap;
}`

think that for building foundation with xy-grid system we need only to include:
`@include foundation-xy-grid-classes;
@include foundation-flex-classes;`

so, the docs are wrong about xy-grid include, just check what foundation-everything is doing :)

**Comment by brettsmason** [comment]
Yep that looks like an error in the docs, you are right it shouldn't be listed at the top and should just be the classes mixin. @kball or @IamManchanda could one if you remove that from the top of the docs page? I'll be home and back to normal from Sunday :smile:

**Comment by bgarrant** [comment]
So to use XY Grid and flex classes it should be this correct?

```
@include foundation-global-styles;
@include foundation-xy-grid-classes;
// @include foundation-grid;
@include foundation-flex-grid;
@include foundation-flex-classes;
@include foundation-typography;
@include foundation-forms;
@include foundation-button;
@include foundation-accordion;
@include foundation-accordion-menu;
@include foundation-badge;
@include foundation-breadcrumbs;
@include foundation-button-group;
@include foundation-callout;
@include foundation-card;
@include foundation-close-button;
@include foundation-menu;
@include foundation-menu-icon;
@include foundation-drilldown-menu;
@include foundation-dropdown;
@include foundation-dropdown-menu;
@include foundation-responsive-embed;
@include foundation-label;
@include foundation-media-object;
@include foundation-off-canvas;
@include foundation-orbit;
@include foundation-pagination;
@include foundation-progress-bar;
@include foundation-slider;
@include foundation-sticky;
@include foundation-reveal;
@include foundation-switch;
@include foundation-table;
@include foundation-tabs;
@include foundation-thumbnail;
@include foundation-title-bar;
@include foundation-tooltip;
@include foundation-top-bar;
@include foundation-visibility-classes;
// @include foundation-float-classes;
```

**Comment by bgarrant** [comment]
can I also comment out the `@include foundation-flex-grid;`?

**Comment by brettsmason** [comment]
@bgarrant yep comment that out too as the flex grid is the legacy flexbox grid which you don't need. And yep looks fine otherwise.

**Comment by bgarrant** [comment]
Works great now. Just need to fix the docs. Also the setup example for Sass does not have the new XY class entry in the list in the docs.

**Comment by bgarrant** [comment]
The component list needs updating on this page http://foundation.zurb.com/sites/docs/sass.html

**Comment by IamManchanda** [comment]
This seems fixed in #10307 ... I am closing this!
